Ongoing contribution for SIPM-SDG

inquisitor101 opened the pull request:
1. Added (time-independent) RHS terms in explicit time stepping. Some that don't, report an error, in case used (e.g. ARS22, ARS33...).
2. added condition number estimate function.
3. added mass inflow conditions in SDG.
4. added Roe + upwind (Vijayasundaram) in SDG.
5. modified the CFL estimate, for a more tighter one (particularly for nPoly=0 is exact).
6. comment clean-ups.
7. code clean-ups (e.g. removing redundant code).
8. removed dirichlet BCs in SDG.

Commit:
[WIP]: IMEX for convection HDG-SDG seems working.

Commit:
[FIX]: corrected RHS comment.

Commit:
[FIX]: changed isvalid_imex_schemes to is_valid_imex_schemes, for consistency.

Commit:
[FIX]: commented redraw functions, since they do not work with netgen atm.

Commit:
[FIX]: added comment indicating the need for future changes in the output.

Commit:
[FIX]: fixed a bug in the solve_stage function call.

Commit:
[FIX]: added comment that the dirk34_ldd cannot be used as an IMEX-pair scheme.

Commit:
[FIX]: changed the time_levels from the nth state, to the (n+1)th.

Commit:
[ADD]: added a working version of a mixed-SIPM HDG-SDG IMEX. Needs cleaning before PR.

inquisitor101 committed:
[ADD]: finished with a IMEX, SDG-HDG NS formulation for prototyping purposes.

inquisitor101 committed:
Merge branch 'feature_gis_imex_nse' into develop

inquisitor101 committed:
[FIX]: changed the comments about averaging.

inquisitor101 committed:
Merge remote-tracking branch 'upstream/develop' into develop

inquisitor101 committed:
[WIP]: bug fixes, viscous formulations for farfield/outflow + addition of Roe and upwind solvers in SDG.

inquisitor101 committed:
[WIP]: reworked the interface condition in HDG + some changes in SDG class and some wip test cases for validation.

inquisitor101 committed:
Merge remote-tracking branch 'upstream/develop' into develop

inquisitor101 committed:
[WIP]: verification of DG is in progress.

Commit:
[WIP]: deleted unnecessary functions, added static inflow and characteristic BCs for SDG + some cleanups.

Commit:
Merge remote-tracking branch 'upstream/develop' into develop

Commit:
reverted the changes in the energy term of the diffusion matrices (it is correct).

Commit:
removed dirichlet BC from SDG + cleaned up some comments.

Commit:
added rhs term to CRK4 (not tested) + error messages for remaining explicit steppers without rhs.

JanEllmenreich commented on dream/solver.py:
```diff
@@ -484,6 +484,25 @@ def set_solver_documentation(self, doc: str):
+        raise NotImplementedError("Overload this function in a derived class.")
+
     def solve(self, reassemble: bool = True):
+        
```
Please do not add this

inquisitor101 replied:
done.

JanEllmenreich commented on examples/.DS_Store:
don't add binaries please

inquisitor101 replied:
done (in the gitignore)

JanEllmenreich commented on examples/compressible/.DS_Store:
same

inquisitor101 replied:
done

Commit:
On branch develop
Your branch is up to date with 'origin/develop'.

Changes to be committed:
      modified:   .gitignore
      modified:   dream/solver.py
      deleted:    examples/test_multizone/compare/multizone/output/.DS_Store
      deleted:    examples/test_multizone/compare/multizone/output/inviscid_structured/hdg_ees.pvd
      deleted:    examples/test_multizone/compare/multizone/output/inviscid_structured/sdg_ees.pvd
      deleted:    examples/test_multizone/compare/multizone/output/viscous_structured/hdg_nse.pvd
      deleted:    examples/test_multizone/compare/multizone/output/viscous_structured/sdg_nse.pvd
      deleted:    examples/test_multizone/compare/singlezone/output/.DS_Store
      deleted:    examples/test_multizone/compare/singlezone/output/inviscid_structured/hdg_ees.pvd
      deleted:    examples/test_multizone/compare/singlezone/output/inviscid_structured/sdg_ees.pvd
      deleted:    examples/test_multizone/compare/singlezone/output/viscous_structured/hdg_nse.pvd
      deleted:    examples/test_multizone/compare/singlezone/output/viscous_structured/sdg_nse.pvd

Commit:
On branch develop
Your branch is up to date with 'origin/develop'.

Changes to be committed:
	deleted:    .DS_Store
	deleted:    examples/.DS_Store
	deleted:    examples/compressible/.DS_Store
	deleted:    examples/test_multizone/.DS_Store
	deleted:    examples/test_multizone/compare/.DS_Store
	deleted:    examples/test_multizone/compare/multizone/.DS_Store
	deleted:    examples/test_multizone/compare/singlezone/.DS_Store
